src/Workspaces/CSharp/Portable/Utilities/CSharpUseImplicitTypeHelper.cs

```diff
@@ -259,9 +259,31 @@ protected override bool ShouldAnalyzeForEachStatement(ForEachStatementSyntax for
             }
 
             // variables declared using var cannot be used further in the same initialization expression.
-            if (initializer.DescendantNodesAndSelf()
-                    .Where(n => (n as IdentifierNameSyntax)?.Identifier.ValueText.Equals(identifier.ValueText) == true)
-                    .Any(n => semanticModel.GetSymbolInfo(n, cancellationToken).Symbol?.IsKind(SymbolKind.Local) == true))
+            bool isReusedInInitializer(SyntaxNode n)
```

Comment:
💡 Naming conventions are coming soon for this, but local functions should be PascalCase.